src/commit.c

```diff
@@ -382,41 +383,48 @@ int git_commit_amend(
 	return error;
 }
 
-int git_commit__parse(void *_commit, git_odb_object *odb_obj)
+int git_commit__parse_commit(git_commit *commit, git_odb_object *odb_obj, git_commit__parse_flags flags)
```

Comment:
I don't love the name, it's a bit redundant.  How about `git_commit__parse_ext`, which is in keeping with other functions that take "extended" argument lists.